### PR TITLE
added beam monitor data, addnl log parameters (no ticket)

### PR DESCRIPTION
**Pull request by geishm-ansto**
Updated the loader to integrate a beam monitor that a added to the EMU instrument at ANSTO. Also added a couple of log parameters to the loaded workspace. 

**To test:**

A new datafile was loaded and the unit test was updated to test the new feature. There are no manual test required for this change. 

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?
- Are the release notes saved in a separate file, using Issue or PR number for file name and in the correct location?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.

### CODE REVIEW

File: Framework/DataHandling/test/LoadEMUauTest.h [hunk 1]
```diff
@@ -83,10 +83,66 @@ class LoadEMUauTest : public CxxTest::TestSuite {
...
+    // check number of histograms and the sum
+    TS_ASSERT_EQUALS(output->getNumberHistograms(), 6592);
+    TS_ASSERT_EQUALS(output->getNumberEvents(), 3135);
+
+    // check the beam monitor rates
+    TS_ASSERT_DELTA(logpm("BeamMonitorRate"), 1482.76, 1.0e-2);
+    TS_ASSERT_DELTA(logpm("BeamMonitorBkgRate"), 1.01, 1.0e-2);
+    TS_ASSERT_EQUALS(logpmi("MonitorCounts"), 27510);
+
+    // repeat with the beam spectrum included
```
Comment (gemmaguest) [inline Framework/DataHandling/test/LoadEMUauTest.h:133]: Could this be refactored into a separate test? This seems to be doing quite a lot for a unit test.

File: Framework/DataHandling/src/LoadEMU.cpp [hunk 2]
```diff
@@ -289,6 +294,29 @@ double maskedStdev(const std::vector<double> &vec, const std::vector<bool> &mask
   return std::sqrt(sum / static_cast<double>(count));
 }
 
+// calculate a running average for a given half window size assuming the data is wrapped
+std::vector<double> runningAverage(const std::vector<size_t> &data, size_t halfWindow) {
+  const auto N = data.size();
+  const size_t TN = 2 * halfWindow + 1;
```
Comment (gemmaguest) [inline Framework/DataHandling/src/LoadEMU.cpp:300]: I found it a bit hard to follow the logic in this function. Is it worth added some more comments and/or descriptive variable names?

File: Framework/DataHandling/src/LoadEMU.cpp [hunk 3]
```diff
@@ -687,9 +759,21 @@ template <typename FD> void LoadEMU<FD>::exec(const std::string &hdfFile, const
   bool saveAsTOF = !Base::getProperty(RawDopplerTimeStr);
   bool loadAsTOF = !m_calibrateDoppler && saveAsTOF;
   EMU::EventAssigner eventAssigner(roi, detMapIndex, PIXELS_PER_TUBE, framePeriod, gatePeriod, timeBoundary,
-                                   directLimits, analysedLimits, convertTOF, eventVectors, start_nanosec, loadAsTOF);
+                                   directLimits, analysedLimits, convertTOF, eventVectors, start_nanosec, loadAsTOF,
+                                   includeBM);
   EMU::loadEvents(prog, "loading neutron events (TOF)", eventFile, eventAssigner);
 
+  // determine the minimum and maximum beam rate per sec
+  auto filteredBM = runningAverage(eventAssigner.beamMonitorCounts(), 5);
```
Comment (gemmaguest) [inline Framework/DataHandling/src/LoadEMU.cpp:767]: It's not clear to me why the hard-coded value `5` is used here - is it worth adding a comment why that value is chosen?

**Review by gemmaguest — CHANGES_REQUESTED** [review]
The code looks good and the new tests look thorough :+1:. Just a few minor queries / suggestions, although this might all be clear to local developers in which case I think it's probably fine.

File: Framework/DataHandling/test/LoadEMUauTest.h (hunk 1 above)
Comment (geishm-ansto) [reply Framework/DataHandling/test/LoadEMUauTest.h:133]: The test is specific to this instrument and I have added a detailed explanation are the start of the test.

File: Framework/DataHandling/src/LoadEMU.cpp (hunk 2 above)
Comment (geishm-ansto) [reply Framework/DataHandling/src/LoadEMU.cpp:300]: I've added a detailed explanation of the filter to the code base and changed the variable name.

File: Framework/DataHandling/src/LoadEMU.cpp (hunk 3 above)
Comment (geishm-ansto) [reply Framework/DataHandling/src/LoadEMU.cpp]: Added an explanation for the choice of the half window size.

**Review by gemmaguest — APPROVED** [review]
Thanks for adding the comments - that makes things much clearer for me :+1:
